src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -9595,6 +9595,12 @@ void Unit::RemoveFromWorld()
 
     if (IsInWorld())
     {
+        if (IsAreaSpiritHealer())
+        {
+            if (Creature* creature = ToCreature())
+                creature->SummonGraveyardTeleporter();
```

Comment:
This seems like a typo? Summoning on RemoveFromWorld doesn't seem safe, at bg destruction this will be also called